linux/devdoc/threadpool_linux_requirements.md

```diff
@@ -0,0 +1,330 @@
+
+**SRS_THREADPOOL_LINUX_11_057: [** `threadpool_work_func` shall decrement the threadpool semaphore with a time limit for 2 seconds. **]**
+
+**SRS_THREADPOOL_LINUX_11_058: [** `threadpool_work_func` shall acquire the shared SRW lock by calling `srw_lock_acquire_shared`. **]**
+
+**SRS_THREADPOOL_LINUX_11_059: [** `threadpool_work_func` shall get the current task array size and next waiting task consume index. **]**
+
+**SRS_THREADPOOL_LINUX_11_060: [** If consume index has task state `TASK_WAITING`, `threadpool_work_func` shall set the task state to `TASK_WORKING`. **]**
+
+**SRS_THREADPOOL_LINUX_11_061: [** `threadpool_work_func` shall initialize `task_func` and `task_param` and then set the task state to `TASK_NOT_USED`. **]**
+
+**SRS_THREADPOOL_LINUX_11_062: [** `threadpool_work_func` shall release the shared SRW lock by calling `srw_lock_release_shared`. **]**
+
+**SRS_THREADPOOL_LINUX_11_063: [** If `task_param` is not `NULL`, `threadpool_work_func` shall execute it with parameter `task_param`. **]**
+
+### reallocate_threadpool_array
+
+```C
+static int reallocate_threadpool_array(THREADPOOL* threadpool);
+```
+
+`reallocate_threadpool_array` resize the task array size by 2.
+
+**SRS_THREADPOOL_LINUX_11_064: [** `reallocate_threadpool_array` shall acquire the SRW lock in exclusive mode by calling `srw_lock_acquire_exclusive`. **]**
+
+**SRS_THREADPOOL_LINUX_11_065: [** `reallocate_threadpool_array` shall get the current size of task array. **]**
+
+**SRS_THREADPOOL_LINUX_11_067: [** Otherwise, `reallocate_threadpool_array` shall double the current task array size and return zero in success. **]**
+
+**SRS_THREADPOOL_LINUX_11_068: [** `reallocate_threadpool_array` shall realloc the memory used for the doubled array items and on success return a non-NULL handle to it. **]**
+
+**SRS_THREADPOOL_LINUX_11_069: [** If any error occurs, `reallocate_threadpool_array` shall fail and return a non-zero value. **]**
+
+**SRS_THREADPOOL_LINUX_11_070: [** `reallocate_threadpool_array` shall initialize every task item in the new task array with `task_func` and `task_param` set to `NULL` and `task_state` set to `TASK_NOT_USED`. **]**
+
+**SRS_THREADPOOL_LINUX_11_071: [** `reallocate_threadpool_array` shall remove any gap in the task array. **]**
+
+**SRS_THREADPOOL_LINUX_11_072: [** `reallocate_threadpool_array` shall reset the `consume_idx` and `insert_idx` to -1 after resize the task array. **]**
+
+**SRS_THREADPOOL_LINUX_11_073: [** `reallocate_threadpool_array` shall release the SRW lock by calling `srw_lock_release_exclusive`. **]**
+
+### internal_close
+
+```C
+static void internal_close(THREADPOOL_HANDLE threadpool);
+```
+
+`internal_close` close the threadpool.
+
+**SRS_THREADPOOL_LINUX_11_077: [** If `current_state` is not `THREADPOOL_STATE_NOT_OPEN`, `internal_close` shall return. **]**
+
+**SRS_THREADPOOL_LINUX_11_078: [** `internal_close` shall wait and close every thread in `threadpool` by calling `ThreadAPI_Join`. **]**
```

Comment:
typically these specs go in the MOCKABLE(..._close) function specs.

And we have a spec in destroy saying something like "shall behave as if close was called"

As it is, this is a static function not referenced by any other spec